src/serve.js

```diff
@@ -1,143 +1,120 @@
+   * If it does it will copy package.json and gulpfile.js from the templates folder
+   * if needed.
+   */
+  ServeTask.assureCordovaProject = function(projectPath) {
+    var deferred = Q.defer();
+
+    fs.exists(path.join(projectPath, 'www'), function(exists) {
+      if (!exists) {
+        deferred.reject('Directory doesn\'t contain a www/ folder.');
+      } else {
+        var httpServerBin = path.join(__dirname, 'serve', 'node_modules', '.bin', 'http-server');
+
+        fs.exists(httpServerBin, function(exists) {
+          if (exists) {
+            deferred.resolve();
+          } else {
+            util.print('Installing packages. Please wait. This might take a couple of minutes.\n');
+
+            var npmProcess = exec('npm install --loglevel error', {
+              cwd: path.join(__dirname, 'serve')
+            });
+
+            npmProcess.stdout.on('data', util.print);
+            npmProcess.stderr.on('data', util.err);
+            npmProcess.on('exit', function(code) {
+              code === 0 ? deferred.resolve() : deferred.reject('Failed installing packages.');
+            });
+          }
+        });
+      }
+    });
+
+    return deferred.promise;
+  };
+
+  ServeTask.run = function(taskName) {
+    this.assureCordovaProject(process.cwd()).then(
+      function() {
+        var fixLog = function(data) {
+          return data.toString()
+            .split('\n')
+            .filter(function(item) {
+              return item !== '';
+            })
+            .map(function(item) {
+              return item + '\n';
+            });
+        };
+
+        var removeSpecialChars = function(msg) {
+          return msg.replace(/\u001b\[.*?m/g, '')
+            .replace(/[^\w\s%/]/gi, '')
+            .replace(/[\n\s\t\r]+/gi, '');
+        }
 
-          var npmProcess = exec('npm install --loglevel error', {cwd: path.join(__dirname, 'serve')});
+        var childProcessBin;
+        var childProcess;
+        var childProcessColor = 'cyan';
 
-          npmProcess.stdout.on('data', util.print);
-          npmProcess.stderr.on('data', util.err);
-          npmProcess.on('exit', function(code) {
-            code === 0 ? deferred.resolve() : deferred.reject('Failed installing packages.');
+        if (monaca.isTranspileEnabled(process.cwd())) {
+          // Webpack Route
+          childProcessBin = monaca.getWebpackDevServerBinPath();
+          childProcess = exec(childProcessBin + ' --progress --config ' + path.join(process.cwd(), 'webpack.dev.config.js') + (argv.port ? ' --port ' + argv.port : ''));
+        } else {
+          console.log('Launching HTTP Server');
+          // HTTP Server Route
+          childProcessBin = path.join(__dirname, 'serve', 'node_modules', 'http-server', 'bin', 'http-server');
+          childProcess = exec('node' + ' ' + childProcessBin + ' ' + path.join(process.cwd(), 'www') + ' -c-1 ' + (argv.open ? ' -o ' : '') + ' -p ' + (argv.port || 8000), {
```

Comment:
This might work correctly, but in general you'd better avoid putting `node` when running a binary file.